packages/core/src/storage/domains/observability/inmemory.ts

```diff
@@ -0,0 +1,175 @@
+import { ErrorCategory, ErrorDomain, MastraError } from '../../../error';
+import { TABLE_AI_SPANS } from '../../constants';
+import type { AISpanRecord, AITraceRecord, AITracesPaginatedArg, PaginationInfo } from '../../types';
+import type { StoreOperations } from '../operations';
+import { ObservabilityStorage } from './base';
+
+export type InMemoryObservability = Map<string, AISpanRecord>;
+export class ObservabilityInMemory extends ObservabilityStorage {
+  operations: StoreOperations;
+  collection: InMemoryObservability;
+
+  constructor({ collection, operations }: { collection: InMemoryObservability; operations: StoreOperations }) {
+    super();
+    this.collection = collection;
+    this.operations = operations;
+  }
+
+  async createAISpan(span: AISpanRecord): Promise<void> {
+    this.validateCreateAISpan(span);
+    const id = this.generateId(span);
+    this.collection.set(id, span);
+  }
+
+  private validateCreateAISpan(record: AISpanRecord): void {
+    if (!record.spanId) {
+      throw new MastraError({
+        id: 'OBSERVABILITY_BATCH_CREATE_AI_SPAN_SPAN_ID_REQUIRED',
```

Comment:
do we often use these huge error IDs?

could this just be `OBSERVABILITY_SPAN_ID_REQUIRED`?